.github/workflows/lint-bpf-checks.yaml

```diff
@@ -124,3 +124,39 @@ jobs:
           fields: repo,message,commit,author,action,eventName,ref,workflow,job,took # selectable (default: repo,message)
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+  bpf_tests:
+    needs: check_changes
+    if: ${{ needs.check_changes.outputs.bpf-tree == 'true' || needs.check_changes.outputs.bpf-tests-runner == 'true' }}
+    name: BPF unit/integration Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a
+        with:
+          go-version: 1.18.4
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@0865c47f36e68161719c5b124609996bb5c40129
+        with:
+          path: $HOME/.clang
+          key: llvm-10.0
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@c538b5e281d5fc40848a3a62636a3a2b6f5a1cfa
+        with:
+          version: "10.0"
+          directory: $HOME/.clang
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+      - name: Checkout code
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - name: Run BPF tests with code coverage reporting
+        run: |
+          make -C test run_bpf_tests COVER=1 || (echo "Run 'make -C test run_bpf_tests COVER=1' locally to investigate failures"; exit 1)
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: bpf-code-coverage-report
+          path: bpf-coverage.html
```

Comment:
It seems we're missing the slack notification step like the previous jobs? Is that deliberate?

Reply:
I have never received a slack notification regarding tests, I only get the GH notifications. So I didn't copy it since I can't test it.

I gather from this comment that slack notification are supposed to work. Is there a special channel for these or do I need to link my GH account somehere?

Reply:
I *suspect* that it's supposed to show up here: https://isovalent.slack.com/archives/C0232ANT40G

(I don't actually use it either, I'm not sure who introduced it or whether they care in this instance. I just noticed this job was different from the previous one).

Reply:
Tracked these notification back to #16218, which indeed seems to post these messages to the mentioned slack channel. I added the step back in. It doesn't seem to trigger on force-pushes though, not sure if that is a feature or a bug.

Reply:
Probably the workflow changes are not being tested in this PR since the PR is opened from your repo and not the cilium/cilium repo..? Otherwise someone could just submit a change to the workflows to print out all the secrets and compromise our supply chain.

Reply:
Ah no I'm probably mistaken here, I think the secrets are protected by a github "environment" which is not available for PRs like this.